junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/MutableExtensionRegistry.java

```diff
@@ -52,7 +52,8 @@ public class MutableExtensionRegistry implements ExtensionRegistry, ExtensionReg
 		new TimeoutExtension(), //
 		new RepeatedTestExtension(), //
 		new TestInfoParameterResolver(), //
-		new TestReporterParameterResolver()));
+		new TestReporterParameterResolver(), //
+		new AutoCloseExtension()));
```

Comment:
The `@ExtendWith(AutoCloseExtension.class)` approach doesn't work atm thus I'll leave the auto-registration in `MutableExtensionRegistry`.  
I think it can be changed to the on demand approach later as well.

Reply:
Please see my other comments about moving the extension to the `junit-jupiter-api` module.